Automattic/woocommerce-payments
# Advanced fraud tools section

Fixes Automattic/woocommerce-payments-server#2797

#### Changes proposed in this Pull Request

This PR adds a new screen to WCPay, that holds the advanced fraud protection settings, designed and proposed in WIM5w2Y10SCNBFdMLSVftj-fi-1300%3A194524. Also contains the list item enabling the user to switch to the advanced protection level, and the button that sends the user to the advanced protection level configuration page on the payments settings page. 









### The advanced protection level selection list item:
<img width="1053" alt="image" src="https://user-images.githubusercontent.com/3295/223512247-3528fc4d-9cc3-45a6-a9a2-597612e34da0.png">

### The page that opens after clicking the "Configure" button:
<img width="963" alt="image" src="https://user-images.githubusercontent.com/3295/223512496-290e920c-0d70-409f-97b1-42999e6d131d.png">

Also this PR contains a new "Amount input" component that's used in the "Purchase price threshold" rule, which allows the user enter only a float in it.

<img width="530" alt="image" src="https://user-images.githubusercontent.com/3295/223512883-b44cde6b-d543-4757-bc29-538f22ead1d2.png">


#### Testing instructions





* Checkout this branch. 
* (Optional) Nothing is changed on `package.json`, but to be sure, you can run `npm ci` to get the matching packages on the `package-lock.json` file.
* Rebuild the scripts with `npm run build:client`
* Enable the fraud protection feature by running `npm run wp option update wcpay_fraud_protection_settings_active 1`
* Go to WCPay Settings, and you should see the Fraud Protection section on the end of the form, before the Advanced link.
* Check if the section contains the "Advanced" checkbox, with the description matching the Figma design. 
* Check if the "Configure" button is only enabled when the advanced protection level is selected.
* Check if the "Configure" button when clicked sends you to the Advanced protection settings page.
* Check if the design of Advanced protection settings matches with the Figma design (Don't bother about the container width's and text breaking on different parts, because that's designed for different screen sizes). 
* Check if enabling/disabling some settings are saved when you click the "Save Changes" button, and not immediately saved by changing the values.
* Check the save isn't triggered when there are some validation errors on forms containing extra fields, such as "Order velocity", "Purchase price threshold" and "Order item count" rules. The warnings will display a yellow box, and the errors will display a red box as stated in the figma. 
* Try to break things. 

-------------------


**Post merge**
